Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -37,34 +36,55 @@ bool LogFileReader::readAll(const QString& fileName)
         if (!parseLine(line, batteryData))
         {
             qDebug() << "Error while parsing" << line;
-            // return false;
+             return false;
         }
         else
         {
-            // This is how to send out a signal in QT using the emit keyword.
-            // This line notifies the classes listening to this signal
-            // that battery data has been received.
             emit batteryDataReceived(batteryData);
         }
     }
 
     return true;
 }
 
-/* File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
- * Need to implement error checking for the correct number of values and
- * that the conversion from string to double is sucessful.*/
+
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
     QStringList sections = line.split(BATDATA_DELIMITER);
 
+    if (sections.length() < COLUMNS)
```

Comment:
As a corollary of my comment below, we can just check if sections.length() is equal to `COLUMNS`.